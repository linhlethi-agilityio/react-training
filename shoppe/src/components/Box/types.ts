import { TUnitValue } from '@type/unitValue';
import { HTMLAttributes, ReactNode } from 'react';
import { TStyleUnit } from '@helper/transformNumber';

export interface IBoxProps extends HTMLAttributes<HTMLDivElement> {
  children: ReactNode;
  mTop?: TUnitValue;
  mLeft?: TUnitValue;
  mRight?: TUnitValue;
  mBottom?: TUnitValue;
  pTop?: TUnitValue;
  pLeft?: TUnitValue;
  pRight?: TUnitValue;
  pBottom?: TUnitValue;
  maxWidth?: TUnitValue;
  unitBox?: TStyleUnit;
}
