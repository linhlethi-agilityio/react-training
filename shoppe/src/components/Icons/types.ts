import { SVGProps } from 'react';

export interface IIconProps extends SVGProps<SVGSVGElement> {
  size?: number;
  color?: string;
}
