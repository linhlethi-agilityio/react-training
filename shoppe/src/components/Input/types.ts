import { BaseVariant } from '@type/variant';
import { InputHTMLAttributes } from 'react';
import { TStyleUnit, TWidthValue } from '@helper/transformNumber';

export interface IInputProps extends InputHTMLAttributes<HTMLInputElement> {
  inputWidth?: number | TWidthValue;
  unitInput?: TStyleUnit;
  error?: boolean;
  variant: BaseVariant;
}
