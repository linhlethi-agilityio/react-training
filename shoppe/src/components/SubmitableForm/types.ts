import { IIconProps } from '@components/Icons/types';
import { BaseVariant } from '@type/variant';
import { ChangeEvent, FunctionComponent, MemoExoticComponent } from 'react';
import { TWidthValue } from '@helper/transformNumber';

export interface IFormControlProps {
  className?: string;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  onSubmit?: () => void;
  IconComponent?: MemoExoticComponent<FunctionComponent<IIconProps>>;
  placeholderInput: string;
  formWidth?: number | TWidthValue;
  inputVariant?: BaseVariant;
  iconSize?: number;
  value: string;
}
