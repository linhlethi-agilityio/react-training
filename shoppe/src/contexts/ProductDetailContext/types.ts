import { IProduct } from '@type/product';
import { ReactNode } from 'react';

export interface IProductDetailContext {
  state: IProductDetailState;
  actions: {
    getProductRequest: () => void;
    getProductRequestSuccess: (response: IProduct) => void;
    getProductRequestFail: (errorMessage: string) => void;
  };
}

export interface IContextProviderProps {
  children: ReactNode;
}

export interface IProductDetailState {
  product: Partial<IProduct>;
  errorMessage: null | string;
  loading: boolean;
}
