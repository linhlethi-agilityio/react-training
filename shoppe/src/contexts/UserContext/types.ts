import { IUser } from '@type/user';
import { ReactNode } from 'react';

export interface IUserContext {
  state: IUserState;
  actions: {
    setLoading: (loading: boolean) => void;
    postLoginRequestSuccess: (token: string, user: IUser) => void;
    setErrorMessage: (message: string) => void;
    logout: () => void;
    onChangeInput: (value: Partial<IUser>) => void;
  };
}

export interface IContextProviderProps {
  children: ReactNode;
}

export interface IUserState {
  userInput: Partial<IUser>;
  user: null | IUser;
  token: null | string;
  loading: boolean;
  errorMessage: null | string;
}
