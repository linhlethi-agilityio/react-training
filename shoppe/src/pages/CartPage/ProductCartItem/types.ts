import { ICartProduct } from '@type/product';

export interface IProductCardProps {
  className?: string;
  data: ICartProduct;
  minQuantity: number;
  maxQuantity: number;
  quantity: number;
  handleDecrease: () => void;
  handleIncrease: () => void;
  handleRemoveProduct: () => void;
}
