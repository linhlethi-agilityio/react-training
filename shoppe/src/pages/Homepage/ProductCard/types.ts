import { IProduct } from '@type/product';

export interface IProductCardProps {
  className?: string;
  data: IProduct;
}
