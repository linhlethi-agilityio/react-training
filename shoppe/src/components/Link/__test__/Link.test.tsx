import { StyledLink, StyledNavigationLink } from '@components/Link';
import { render } from '@testing-library/react';
import { ILinkProps } from '../types';
import { BrowserRouter } from 'react-router-dom';

const StyledLinkWithRouter = (props: ILinkProps) => {
  return (
    <BrowserRouter>
      <StyledLink {...props} />
    </BrowserRouter>
  );
};

const StyledNavigationLinkWithRouter = (props: ILinkProps) => {
  return (
    <BrowserRouter>
      <StyledNavigationLink {...props} />
    </BrowserRouter>
  );
};

describe('testing Link component', () => {
  it('Link component matches snapshot', () => {
    const { asFragment } = render(<StyledLinkWithRouter to="/">link</StyledLinkWithRouter>);

    expect(asFragment()).toMatchSnapshot();
  });

  it('NavigationLink component matches snapshot', () => {
    const { asFragment } = render(
      <StyledNavigationLinkWithRouter to="/">link</StyledNavigationLinkWithRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render disabled props', () => {
    const { asFragment } = render(
      <StyledLinkWithRouter disabled={true} to="/">
        link
      </StyledLinkWithRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render link correctly', () => {
    const { getByRole } = render(<StyledLinkWithRouter to="/">link</StyledLinkWithRouter>);

    const link = getByRole('link') as HTMLLinkElement;

    expect(link).toBeTruthy();
    expect(link.href).toBe('http://localhost/');
    expect(link.innerHTML).toBe('link');
  });
});
