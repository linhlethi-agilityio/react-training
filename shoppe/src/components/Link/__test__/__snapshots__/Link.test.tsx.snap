// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Link component Link component matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="sc-beqWaB iEMdCi"
    href="/"
  >
    link
  </a>
</DocumentFragment>
`;

exports[`testing Link component NavigationLink component matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="sc-beqWaB iEMdCi sc-gueYoa cqdWIK"
    href="/"
  >
    link
  </a>
</DocumentFragment>
`;

exports[`testing Link component should render disabled props 1`] = `
<DocumentFragment>
  <a
    class="sc-beqWaB fmIVOE"
    disabled=""
    href="/"
  >
    link
  </a>
</DocumentFragment>
`;
